File: 19. Remove Nth Node From End of List.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
/* Fast and slow pointers. */
struct ListNode* removeNthFromEnd(struct ListNode* head, int n) {
    int i;
    struct ListNode **fast, **slow;

    i = 1;
    fast = slow = &head;
    while(i < n){
        fast = &(*fast)->next;
        i++;
    }
    while((*fast)->next){
        fast = &(*fast)->next;
        slow = &(*slow)->next;
    }
    *slow = (*slow)->next;
    return head;
}

File: Python3/19. Remove Nth Node From End of List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def removeNthFromEnd(self, head, n):
        """
        :type head: ListNode
        :type n: int
        :rtype: ListNode
        """
        dummy, dummy.next = ListNode(0), head
        fast = slow = dummy

        for _ in range(n):
            fast = fast.next
        while fast.next:
            slow, fast = slow.next, fast.next
        slow.next = slow.next.next
        return dummy.next
